deployability/modules/generic/schema_validator.py

```diff
@@ -0,0 +1,72 @@
+import jsonschema
+import json
+
+from jsonschema.exceptions import ValidationError
+from pathlib import Path
+from ruamel.yaml import YAML
+
+from modules.generic.logger import Logger
+
+
+class SchemaValidator:
+    """
+    A SchemaValidator class that validates a YAML file against a JSON schema.
+
+    Attributes:
+        schema_data (dict): The schema data.
+        yaml_data (dict): The YAML data.
+    """
+
+    def __init__(self, schema: Path | str, to_validate: Path | str):
```

Comment:
Add Docstrings.

Reply:
added in https://github.com/wazuh/wazuh-qa/pull/4897/commits/68769f6a770b751217842d99b45099e051ac2378